.github/workflows/linter.yml

```diff
@@ -0,0 +1,50 @@
+# ---------------------------------------#
+# ---------------------------------------#
+# --- SUPER LINTER - Github Actions -----#
+# ---------------------------------------#
+# ---------------------------------------#
+
+name: Lint Code Base
+#
+# Documentation:
+# https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions
+#
+# Find super linter at:
+# https://github.com/github/super-linter
+
+on:
+  push:
+    paths:
+      # Add specific files / filetypes here.
+      # Use `**` to wildcard through multiply paths
+      - "**.py"
+    # [un]-comment the following line to toggle for push to master
+    # branches-ignore: [master, main]
+
+  # This will only work when this workflow is on default branch
+  # TODO: Uncomment when merging
+  # pull_requests:
+  #   branches: [dev, master, main]
+
+jobs:
```

Comment:
this should be added to the main pipeline, as a stage. We want the pipeline to fail on linting anyways, so that is probably reasonable.